### ISSUE TITLE
Add `log_weight_decay` option to `LearningRateMonitor` callback

### ISSUE THREAD

**Issue by ioangatop** [Lightning-AI/lightning#18117]
### Description & Motivation

In more sophisticated training settings, we usually apply a dedicated weight decay scheduler (for example DINO). Thus, it will be great to monitor these statistics, optionally, from the `LearningRateMonitor` callback.

### Pitch

A solution should look something like the following, but build in the LearningRateMonitor:
```py
"""Learning rate monitor logger callback."""
from typing import Any, Dict, List

from pytorch_lightning import callbacks
from torch import optim


class LearningRateMonitor(callbacks.LearningRateMonitor):
    """Extends the PL version with the ability to log the weight decay."""

    def __init__(
        self,
        logging_interval: str | None = None,
        log_momentum: bool = False,
        log_weight_decay: bool = False,
    ) -> None:
        """Initializes the callback."""
        super().__init__(logging_interval, log_momentum)
        self.log_weight_decay = log_weight_decay

    def _get_lr_momentum_stat(
        self, optimizer: optim.Optimizer, names: List[str]
    ) -> Dict[str, float]:
        """Gathers and returns the learning rate, momentum and weight decay statistics."""
        lr_momentum_wd_stat = {}
        param_groups = optimizer.param_groups
        use_betas = "betas" in optimizer.defaults

        for pg, name in zip(param_groups, names):
            lr = self._extract_lr(pg, name)
            lr_momentum_wd_stat.update(lr)
            momentum = self._extract_momentum(
                param_group=pg, name=name.replace(name, f"{name}-momentum"), use_betas=use_betas
            )
            lr_momentum_wd_stat.update(momentum)
            weight_decay = self._extract_weight_decay(pg, name + "-weight_decay")
            lr_momentum_wd_stat.update(weight_decay)

        return lr_momentum_wd_stat

    def _extract_weight_decay(self, param_group: Dict[str, Any], name: str) -> Dict[str, Any]:
        """Extracts the weight decay statistics from a parameter group."""
        if not self.log_weight_decay:
            return {}

        wd = param_group["weight_decay"]
        return {name: wd}
```

cc @borda

**Comment by awaelchli** [comment]
Thanks @lobantseff for this suggestion. I don't see what could stand in the way of such an addition. A PR for this would be welcomed if you are interested to contribute the feature :)
